astropy/units/equivalencies.py

```diff
@@ -811,3 +811,31 @@ def with_H0(H0=None):
     h100_val_unit = Unit(100/(H0.to_value((si.km/si.s)/astrophys.Mpc)) * astrophys.littleh)
 
     return Equivalency([(h100_val_unit, None)], "with_H0", kwargs={"H0": H0})
+
+
+def cosmological(z, cosmology=None, comoving=False):
+    """
+    Returns the equivalence between angles and cosmological distances.
```

Comment:
Maybe further emphasize that this equivalence really only applies where the expansion dominates over the local gravitational field — aka cosmological scales. 

```suggestion
    Returns the equivalence on cosmological scales between angles and distances.
```